library/psa_crypto.c

```diff
@@ -292,12 +324,14 @@ psa_status_t mbedtls_to_psa_error(int ret)
             return PSA_ERROR_INVALID_SIGNATURE;
         case MBEDTLS_ERR_PK_BUFFER_TOO_SMALL:
             return PSA_ERROR_BUFFER_TOO_SMALL;
+#endif
 
         case MBEDTLS_ERR_PLATFORM_HW_ACCEL_FAILED:
             return PSA_ERROR_HARDWARE_FAILURE;
         case MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED:
             return PSA_ERROR_NOT_SUPPORTED;
 
+#if defined(MBEDTLS_RSA_C)
```

Comment:
It really "feels like" this should be a `PSA_WANT_KEY_TYPE_xxx` but I can't see one that actually works - I'd expect `PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_BASIC` but that doesn't look right

Reply:
Normally the PSA core should not use `MBEDTLS_xxx` symbols, but error conversion is a special case (and arguably should be in a separate file). `MBDETLS_ERR_RSA_xxx` errors come from `rsa.c`, so `MBEDTLS_RSA_C` is the correct guard.

Reply:
> `MBDETLS_ERR_RSA_xxx` errors come from `rsa.c`, so `MBEDTLS_RSA_C` is the correct guard.

Are you okay with the use of the symmetric `PSA_WANT_ALG_xxx` and `PSA_WANT_KEY_TYPE_xxx` guards added then? They are correctly guarding errors that are related to the wanted algorithm and/or key types, but if they should be `MBEDTLS_xxx` guards then they need to be changed

Reply:
@gilles-peskine-arm Thanks for taking the time answer that.

@daverodgman Can we replace the new `PSA_WANT_xxx` guards with the appropriate `MBEDTLS_xxx` ones? Sorry...

Reply:
I agree, I think `MBEDTLS_xxx` makes more sense. I'll update.

Reply:
Strictly speaking, the condition for needing the conversion from `MBEDTLS_ERR_xxx` to a PSA error code is: (this error code is emitted) and (there is PSA code calling the emitting code). The first condition is always an `MBEDTLS_xxx` guard (unless there's a bug lingering somewhere, but with our level of driver support the chances are low these days). Typically `MBEDTLS_xxx_C`, but for MD and ECP they're `MBEDTLS_xxx_LIGHT`. The second condition is more complex, because it's technically possible e.g. to enable `MBEDTLS_ECP_C` and `MBEDTLS_PSA_CRYPTO_C` but not any ECC mechanism in PSA. But those would be very unusual scenarios and we don't need to worry about optimizing them unless somebody complains (and even so we might just shrug it off). Thus, unless there's a good reason, the guard should just be the corresponding `MBEDTLS_xxx_(C|LIGHT)`.